astropy/table/column.py

```diff
@@ -131,25 +80,28 @@ def col_iter_str_vals(col):
 def col_copy(col):
     """
     This is a mixin-safe version of Column.copy() (with copy_data=True).
-
-    Warning: this function is subject to change or removal.
     """
     if isinstance(col, BaseColumn):
         return col.copy()
 
-    if hasattr(col, '_astropy_column_attrs'):
-        col_setattr(col, 'parent_table', None)  # Don't copy weakref to parent table
-    newcol = col.copy() if hasattr(col, 'copy') else deepcopy(col)
+    col.info.parent_table = None  # Don't copy weakref to parent table
 
-    # Copy old attributes.  Even deepcopy above may not get this (e.g. pandas).
-    if (not hasattr(newcol, '_astropy_column_attrs') or
-            newcol._astropy_column_attrs is None):
-        _column_attrs = deepcopy(getattr(col, '_astropy_column_attrs', {}))
-        newcol._astropy_column_attrs = _column_attrs
+    newcol = col.copy() if hasattr(col, 'copy') else deepcopy(col)
+    info_attr = '_info' if hasattr(col.__class__, 'info') else 'info'
+    setattr(newcol, info_attr, col.info.copy())
+    newcol.info._parent_col = weakref.ref(newcol)
 
     return newcol
 
 
+def add_column_info(col):
```

Comment:
Since it is not yet directly needed, my tendency would be to omit this for now, and simply explain what one would have to do in the documentation (maybe with a warning that this is still subject to change). 

If we do keep it, we should raise an exception if `info` is already present.